test/modules/TestResultCoordinator/Settings.cs

```diff
@@ -131,51 +134,34 @@ public override string ToString()
             return $"Settings:{Environment.NewLine}{string.Join(Environment.NewLine, fields.Select(f => $"{f.Key}={f.Value}"))}";
         }
 
-        internal List<ITestReportMetadata> GetReportMetadataList()
+        internal async Task<List<ITestReportMetadata>> GetReportMetadataListAsync(ILogger logger)
```

Comment:
With the "Current" pattern, it feels like the Twin should be obtained and cached in Create() itself.

Reply:
Current pattern is the singleton pattern to ensure only 1 Settings in the process.  I think it is reasonable for this class to provide methods to manage more setting-related data.

If you prefer to change Create method become async, then I need to:
1. Rename Create method to GetOrCreateAsync.
2. Update all usage of Settings.Current to become Settings.GetOrCreateAsync; which will use a simple lock and check whether to return cached object or create a new instance.

Please confirm which implementation you recommended; or suggest another way.  thanks.